src/backend/distributed/commands/create_citus_local_table.c

```diff
@@ -0,0 +1,620 @@
+
+	char *indexName = NULL;
+	foreach_ptr(indexName, indexNameList)
+	{
+		char *shardIndexName = pstrdup(indexName);
+		AppendShardIdToName(&shardIndexName, shardId);
+
+		StringInfo renameCommand = makeStringInfo();
+		appendStringInfo(renameCommand, "ALTER INDEX %s RENAME TO %s;",
+						 indexName, shardIndexName);
+
+		const char *commandString = renameCommand->data;
+
+		Node *parseTree = ParseTreeNode(commandString);
+
+		CitusProcessUtility(parseTree, commandString, PROCESS_UTILITY_TOPLEVEL,
+							NULL, None_Receiver, NULL);
+	}
+}
+
+
+/*
+ * GetExplicitIndexNameList returns a list of index names defined "explicitly"
+ * on the relation with relationId by the "CREATE INDEX" commands. That means,
+ * all the constraints defined on the relation except:
+ *  - primary indexes,
+ *  - unique indexes and
+ *  - exclusion indexes
+ * that are actually applied by the related constraints.
+ */
+static List *
+GetExplicitIndexNameList(Oid relationId)
+{
+	List *indexNameList = NIL;
+
+	int scanKeyCount = 1;
+	ScanKeyData scanKey[1];
+
+	/*
+	 * Set search_path to NIL so that all objects outside of pg_catalog will be
+	 * schema-prefixed. pg_catalog will be added automatically when we call
+	 * PushOverrideSearchPath(), since we set addCatalog to true;
+	 */
+	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
+	overridePath->schemas = NIL;
+	overridePath->addCatalog = true;
+	PushOverrideSearchPath(overridePath);
+
+	Relation pgIndex = heap_open(IndexRelationId, AccessShareLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_index_indrelid,
+				BTEqualStrategyNumber, F_OIDEQ, relationId);
+
+	bool useIndex = true;
+	SysScanDesc scanDescriptor = systable_beginscan(pgIndex, IndexIndrelidIndexId,
+													useIndex, NULL, scanKeyCount,
+													scanKey);
+
+	HeapTuple heapTuple = systable_getnext(scanDescriptor);
+	while (HeapTupleIsValid(heapTuple))
+	{
+		Form_pg_index indexForm = (Form_pg_index) GETSTRUCT(heapTuple);
+
+		if (indexForm->indisprimary || indexForm->indisunique ||
+			indexForm->indisexclusion)
+		{
+			/*
+			 * Skip the indexes that are not implied by explicitly executing
+			 * a CREATE INDEX command.
+			 */
+			heapTuple = systable_getnext(scanDescriptor);
+			continue;
+		}
+
+		Oid indexId = indexForm->indexrelid;
+
+		char *indexName = get_rel_name(indexId);
+
+		indexNameList = lappend(indexNameList, pstrdup(indexName));
+
+		heapTuple = systable_getnext(scanDescriptor);
+	}
+
+	systable_endscan(scanDescriptor);
+	heap_close(pgIndex, AccessShareLock);
+
+	/* revert back to original search_path */
+	PopOverrideSearchPath();
+
+	return indexNameList;
+}
+
+
+/*
+ * CreateShellTableForCitusLocalTable creates the shell table for the citus
+ * local tables by executing the given commands necessary to recreate the
+ * table.
+ */
+static void
+CreateShellTableForCitusLocalTable(List *shellTableDDLEvents)
+{
+	Assert(list_length(shellTableDDLEvents) > 0);
```

Comment:
Actually what I wanted to assert there was if we have commands to recreate the shell table. This is also because we generate ddl events list for shell table by concatenating output of two different functions (For example, it is also fine if `GetForeignConstraintCommandsTableInvolved` returns NIL).

Reply:
Yeah there are two different functions but if we use a single separate function to create shell table ddl commands then we can have the assertion at the end of that function where we are returning the ddl command list, which should be nonempty. It is a minor thing, so feel free to ignore